--- Firmware/src/modules/sensors/sensor_params.c	2015-04-07 12:50:25.906575935 +0530
+++ pandapilot_v4/src/modules/sensors/sensor_params.c	2015-04-09 12:07:09.575942682 +0530
@@ -1,6 +1,6 @@
 /****************************************************************************
- *
- *   Copyright (c) 2012-2015 PX4 Development Team. All rights reserved.
+ *   Copyright (c) 2014 NavStik Development Team. All rights reserved.
+ *   Copyright (c) 2012-2014 PX4 Development Team. All rights reserved.
  *
  * Redistribution and use in source and binary forms, with or without
  * modification, are permitted provided that the following conditions
@@ -36,36 +36,22 @@
  *
  * Parameters defined by the sensors task.
  *
- * @author Lorenz Meier <lorenz@px4.io>
- * @author Julian Oes <julian@px4.io>
- * @author Thomas Gubler <thomas@px4.io>
+ * @author Lorenz Meier <lm@inf.ethz.ch>
+ * @author Julian Oes <joes@student.ethz.ch>
+ * @author Thomas Gubler <thomasgubler@student.ethz.ch>
  */
 
 #include <nuttx/config.h>
 #include <systemlib/param/param.h>
 
 /**
- * ID of the board this parameter set was calibrated on.
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_INT32(CAL_BOARD_ID, 0);
-
-/**
- * ID of the Gyro that the calibration is for.
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_INT32(CAL_GYRO0_ID, 0);
-
-/**
  * Gyro X-axis offset
  *
  * @min -10.0
  * @max 10.0
  * @group Sensor Calibration
  */
-PARAM_DEFINE_FLOAT(CAL_GYRO0_XOFF, 0.0f);
+PARAM_DEFINE_FLOAT(SENS_GYRO_XOFF, 0.0f);
 
 /**
  * Gyro Y-axis offset
@@ -74,7 +60,7 @@
  * @max 10.0
  * @group Sensor Calibration
  */
-PARAM_DEFINE_FLOAT(CAL_GYRO0_YOFF, 0.0f);
+PARAM_DEFINE_FLOAT(SENS_GYRO_YOFF, 0.0f);
 
 /**
  * Gyro Z-axis offset
@@ -83,7 +69,7 @@
  * @max 5.0
  * @group Sensor Calibration
  */
-PARAM_DEFINE_FLOAT(CAL_GYRO0_ZOFF, 0.0f);
+PARAM_DEFINE_FLOAT(SENS_GYRO_ZOFF, 0.0f);
 
 /**
  * Gyro X-axis scaling factor
@@ -92,7 +78,7 @@
  * @max 1.5
  * @group Sensor Calibration
  */
-PARAM_DEFINE_FLOAT(CAL_GYRO0_XSCALE, 1.0f);
+PARAM_DEFINE_FLOAT(SENS_GYRO_XSCALE, 1.0f);
 
 /**
  * Gyro Y-axis scaling factor
@@ -101,7 +87,7 @@
  * @max 1.5
  * @group Sensor Calibration
  */
-PARAM_DEFINE_FLOAT(CAL_GYRO0_YSCALE, 1.0f);
+PARAM_DEFINE_FLOAT(SENS_GYRO_YSCALE, 1.0f);
 
 /**
  * Gyro Z-axis scaling factor
@@ -110,27 +96,8 @@
  * @max 1.5
  * @group Sensor Calibration
  */
-PARAM_DEFINE_FLOAT(CAL_GYRO0_ZSCALE, 1.0f);
-
-/**
- * ID of Magnetometer the calibration is for.
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_INT32(CAL_MAG0_ID, 0);
+PARAM_DEFINE_FLOAT(SENS_GYRO_ZSCALE, 1.0f);
 
-/**
- * Rotation of magnetometer 0 relative to airframe.
- *
- * An internal magnetometer will force a value of -1, so a GCS
- * should only attempt to configure the rotation if the value is
- * greater than or equal to zero.
- *
- * @min -1
- * @max 30
- * @group Sensor Calibration
- */
-PARAM_DEFINE_INT32(CAL_MAG0_ROT, -1);
 
 /**
  * Magnetometer X-axis offset
@@ -139,7 +106,7 @@
  * @max 500.0
  * @group Sensor Calibration
  */
-PARAM_DEFINE_FLOAT(CAL_MAG0_XOFF, 0.0f);
+PARAM_DEFINE_FLOAT(SENS_MAG_XOFF, 0.0f);
 
 /**
  * Magnetometer Y-axis offset
@@ -148,7 +115,7 @@
  * @max 500.0
  * @group Sensor Calibration
  */
-PARAM_DEFINE_FLOAT(CAL_MAG0_YOFF, 0.0f);
+PARAM_DEFINE_FLOAT(SENS_MAG_YOFF, 0.0f);
 
 /**
  * Magnetometer Z-axis offset
@@ -157,433 +124,72 @@
  * @max 500.0
  * @group Sensor Calibration
  */
-PARAM_DEFINE_FLOAT(CAL_MAG0_ZOFF, 0.0f);
+PARAM_DEFINE_FLOAT(SENS_MAG_ZOFF, 0.0f);
 
 /**
  * Magnetometer X-axis scaling factor
  *
  * @group Sensor Calibration
  */
-PARAM_DEFINE_FLOAT(CAL_MAG0_XSCALE, 1.0f);
+PARAM_DEFINE_FLOAT(SENS_MAG_XSCALE, 1.0f);
 
 /**
  * Magnetometer Y-axis scaling factor
  *
  * @group Sensor Calibration
  */
-PARAM_DEFINE_FLOAT(CAL_MAG0_YSCALE, 1.0f);
+PARAM_DEFINE_FLOAT(SENS_MAG_YSCALE, 1.0f);
 
 /**
  * Magnetometer Z-axis scaling factor
  *
  * @group Sensor Calibration
  */
-PARAM_DEFINE_FLOAT(CAL_MAG0_ZSCALE, 1.0f);
+PARAM_DEFINE_FLOAT(SENS_MAG_ZSCALE, 1.0f);
 
-/**
- * ID of the Accelerometer that the calibration is for.
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_INT32(CAL_ACC0_ID, 0);
 
 /**
  * Accelerometer X-axis offset
  *
  * @group Sensor Calibration
  */
-PARAM_DEFINE_FLOAT(CAL_ACC0_XOFF, 0.0f);
+PARAM_DEFINE_FLOAT(SENS_ACC_XOFF, 0.0f);
 
 /**
  * Accelerometer Y-axis offset
  *
  * @group Sensor Calibration
  */
-PARAM_DEFINE_FLOAT(CAL_ACC0_YOFF, 0.0f);
+PARAM_DEFINE_FLOAT(SENS_ACC_YOFF, 0.0f);
 
 /**
  * Accelerometer Z-axis offset
  *
  * @group Sensor Calibration
  */
-PARAM_DEFINE_FLOAT(CAL_ACC0_ZOFF, 0.0f);
+PARAM_DEFINE_FLOAT(SENS_ACC_ZOFF, 0.0f);
 
 /**
  * Accelerometer X-axis scaling factor
  *
  * @group Sensor Calibration
  */
-PARAM_DEFINE_FLOAT(CAL_ACC0_XSCALE, 1.0f);
+PARAM_DEFINE_FLOAT(SENS_ACC_XSCALE, 1.0f);
 
 /**
  * Accelerometer Y-axis scaling factor
  *
  * @group Sensor Calibration
  */
-PARAM_DEFINE_FLOAT(CAL_ACC0_YSCALE, 1.0f);
+PARAM_DEFINE_FLOAT(SENS_ACC_YSCALE, 1.0f);
 
 /**
  * Accelerometer Z-axis scaling factor
  *
  * @group Sensor Calibration
  */
-PARAM_DEFINE_FLOAT(CAL_ACC0_ZSCALE, 1.0f);
-
-/**
- * ID of the Gyro that the calibration is for.
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_INT32(CAL_GYRO1_ID, 0);
-
-/**
- * Gyro X-axis offset
- *
- * @min -10.0
- * @max 10.0
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_GYRO1_XOFF, 0.0f);
+PARAM_DEFINE_FLOAT(SENS_ACC_ZSCALE, 1.0f);
 
-/**
- * Gyro Y-axis offset
- *
- * @min -10.0
- * @max 10.0
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_GYRO1_YOFF, 0.0f);
-
-/**
- * Gyro Z-axis offset
- *
- * @min -5.0
- * @max 5.0
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_GYRO1_ZOFF, 0.0f);
-
-/**
- * Gyro X-axis scaling factor
- *
- * @min -1.5
- * @max 1.5
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_GYRO1_XSCALE, 1.0f);
-
-/**
- * Gyro Y-axis scaling factor
- *
- * @min -1.5
- * @max 1.5
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_GYRO1_YSCALE, 1.0f);
-
-/**
- * Gyro Z-axis scaling factor
- *
- * @min -1.5
- * @max 1.5
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_GYRO1_ZSCALE, 1.0f);
-
-/**
- * ID of Magnetometer the calibration is for.
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_INT32(CAL_MAG1_ID, 0);
-
-/**
- * Rotation of magnetometer 1 relative to airframe.
- *
- * An internal magnetometer will force a value of -1, so a GCS
- * should only attempt to configure the rotation if the value is
- * greater than or equal to zero.
- *
- * @min -1
- * @max 30
- * @group Sensor Calibration
- */
-PARAM_DEFINE_INT32(CAL_MAG1_ROT, -1);
-
-/**
- * Magnetometer X-axis offset
- *
- * @min -500.0
- * @max 500.0
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_MAG1_XOFF, 0.0f);
-
-/**
- * Magnetometer Y-axis offset
- *
- * @min -500.0
- * @max 500.0
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_MAG1_YOFF, 0.0f);
-
-/**
- * Magnetometer Z-axis offset
- *
- * @min -500.0
- * @max 500.0
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_MAG1_ZOFF, 0.0f);
-
-/**
- * Magnetometer X-axis scaling factor
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_MAG1_XSCALE, 1.0f);
-
-/**
- * Magnetometer Y-axis scaling factor
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_MAG1_YSCALE, 1.0f);
-
-/**
- * Magnetometer Z-axis scaling factor
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_MAG1_ZSCALE, 1.0f);
-
-/**
- * ID of the Accelerometer that the calibration is for.
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_INT32(CAL_ACC1_ID, 0);
-
-/**
- * Accelerometer X-axis offset
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_ACC1_XOFF, 0.0f);
-
-/**
- * Accelerometer Y-axis offset
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_ACC1_YOFF, 0.0f);
-
-/**
- * Accelerometer Z-axis offset
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_ACC1_ZOFF, 0.0f);
-
-/**
- * Accelerometer X-axis scaling factor
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_ACC1_XSCALE, 1.0f);
-
-/**
- * Accelerometer Y-axis scaling factor
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_ACC1_YSCALE, 1.0f);
-
-/**
- * Accelerometer Z-axis scaling factor
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_ACC1_ZSCALE, 1.0f);
-
-/**
- * ID of the Gyro that the calibration is for.
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_INT32(CAL_GYRO2_ID, 0);
-
-/**
- * Gyro X-axis offset
- *
- * @min -10.0
- * @max 10.0
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_GYRO2_XOFF, 0.0f);
-
-/**
- * Gyro Y-axis offset
- *
- * @min -10.0
- * @max 10.0
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_GYRO2_YOFF, 0.0f);
-
-/**
- * Gyro Z-axis offset
- *
- * @min -5.0
- * @max 5.0
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_GYRO2_ZOFF, 0.0f);
-
-/**
- * Gyro X-axis scaling factor
- *
- * @min -1.5
- * @max 1.5
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_GYRO2_XSCALE, 1.0f);
-
-/**
- * Gyro Y-axis scaling factor
- *
- * @min -1.5
- * @max 1.5
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_GYRO2_YSCALE, 1.0f);
-
-/**
- * Gyro Z-axis scaling factor
- *
- * @min -1.5
- * @max 1.5
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_GYRO2_ZSCALE, 1.0f);
-
-/**
- * ID of Magnetometer the calibration is for.
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_INT32(CAL_MAG2_ID, 0);
-
-/**
- * Rotation of magnetometer 2 relative to airframe.
- *
- * An internal magnetometer will force a value of -1, so a GCS
- * should only attempt to configure the rotation if the value is
- * greater than or equal to zero.
- *
- * @min -1
- * @max 30
- * @group Sensor Calibration
- */
-PARAM_DEFINE_INT32(CAL_MAG2_ROT, -1);
-
-/**
- * Magnetometer X-axis offset
- *
- * @min -500.0
- * @max 500.0
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_MAG2_XOFF, 0.0f);
-
-/**
- * Magnetometer Y-axis offset
- *
- * @min -500.0
- * @max 500.0
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_MAG2_YOFF, 0.0f);
-
-/**
- * Magnetometer Z-axis offset
- *
- * @min -500.0
- * @max 500.0
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_MAG2_ZOFF, 0.0f);
-
-/**
- * Magnetometer X-axis scaling factor
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_MAG2_XSCALE, 1.0f);
-
-/**
- * Magnetometer Y-axis scaling factor
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_MAG2_YSCALE, 1.0f);
-
-/**
- * Magnetometer Z-axis scaling factor
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_MAG2_ZSCALE, 1.0f);
-
-/**
- * ID of the Accelerometer that the calibration is for.
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_INT32(CAL_ACC2_ID, 0);
-
-/**
- * Accelerometer X-axis offset
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_ACC2_XOFF, 0.0f);
-
-/**
- * Accelerometer Y-axis offset
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_ACC2_YOFF, 0.0f);
-
-/**
- * Accelerometer Z-axis offset
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_ACC2_ZOFF, 0.0f);
-
-/**
- * Accelerometer X-axis scaling factor
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_ACC2_XSCALE, 1.0f);
-
-/**
- * Accelerometer Y-axis scaling factor
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_ACC2_YSCALE, 1.0f);
-
-/**
- * Accelerometer Z-axis scaling factor
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_FLOAT(CAL_ACC2_ZSCALE, 1.0f);
 
 /**
  * Differential pressure sensor offset
@@ -656,25 +262,6 @@
 PARAM_DEFINE_INT32(SENS_BOARD_ROT, 0);
 
 /**
- * PX4Flow board rotation
- *
- * This parameter defines the rotation of the PX4FLOW board relative to the platform.
- * Zero rotation is defined as Y on flow board pointing towards front of vehicle
- * Possible values are:
- *    0 = No rotation
- *    1 = Yaw 45°
- *    2 = Yaw 90°
- *    3 = Yaw 135°
- *    4 = Yaw 180°
- *    5 = Yaw 225°
- *    6 = Yaw 270°
- *    7 = Yaw 315°
- *
- * @group Sensor Calibration
- */
-PARAM_DEFINE_INT32(SENS_FLOW_ROT, 0);
-
-/**
  * Board rotation Y (Pitch) offset
  *
  * This parameter defines a rotational offset in degrees around the Y (Pitch) axis. It allows the user
@@ -682,7 +269,7 @@
  *
  * @group Sensor Calibration
  */
-PARAM_DEFINE_FLOAT(SENS_BOARD_Y_OFF, 0.0f);
+ PARAM_DEFINE_FLOAT(SENS_BOARD_Y_OFF, 0.0f);
 
 /**
  * Board rotation X (Roll) offset
@@ -963,6 +550,13 @@
  * @group Battery Calibration
  */
 PARAM_DEFINE_FLOAT(BAT_V_SCALING, 0.0082f);
+#elif CONFIG_ARCH_BOARD_NAVSTIK_V1
+/**
+ * Scaling factor for battery voltage sensor on NAVSTIK.
+ *
+ * @group Battery Calibration
+ */
+PARAM_DEFINE_FLOAT(BAT_V_SCALING, 0.00388436f);
 #elif CONFIG_ARCH_BOARD_AEROCORE
 /**
  * Scaling factor for battery voltage sensor on AeroCore.
@@ -994,36 +588,6 @@
 
 
 /**
- * RC channel count
- *
- * This parameter is used by Ground Station software to save the number
- * of channels which were used during RC calibration. It is only meant
- * for ground station use.
- *
- * @min 0
- * @max 18
- * @group Radio Calibration
- */
-
-PARAM_DEFINE_INT32(RC_CHAN_CNT, 0);
-
-/**
- * RC mode switch threshold automaic distribution
- *
- * This parameter is used by Ground Station software to specify whether
- * the threshold values for flight mode switches were automatically calculated.
- * 0 indicates that the threshold values were set by the user. Any other value
- * indicates that the threshold value where automatically set by the ground
- * station software. It is only meant for ground station use.
- *
- * @min 0
- * @max 1
- * @group Radio Calibration
- */
-
-PARAM_DEFINE_INT32(RC_TH_USER, 1);
-
-/**
  * Roll control channel mapping.
  *
  * The channel index (starting from 1 for channel 1) indicates
@@ -1190,41 +754,6 @@
  */
 PARAM_DEFINE_INT32(RC_MAP_AUX3, 0);
 
-/**
- * Channel which changes a parameter
- *
- * Can be used for parameter tuning with the RC. This one is further referenced as the 1st parameter channel.
- * Set to 0 to deactivate *
- *
- * @min 0
- * @max 18
- * @group Radio Calibration
- */
-PARAM_DEFINE_INT32(RC_MAP_PARAM1, 0);
-
-/**
- * Channel which changes a parameter
- *
- * Can be used for parameter tuning with the RC. This one is further referenced as the 2nd parameter channel.
- * Set to 0 to deactivate *
- *
- * @min 0
- * @max 18
- * @group Radio Calibration
- */
-PARAM_DEFINE_INT32(RC_MAP_PARAM2, 0);
-
-/**
- * Channel which changes a parameter
- *
- * Can be used for parameter tuning with the RC. This one is further referenced as the 3th parameter channel.
- * Set to 0 to deactivate *
- *
- * @min 0
- * @max 18
- * @group Radio Calibration
- */
-PARAM_DEFINE_INT32(RC_MAP_PARAM3, 0);
 
 /**
  * Failsafe channel PWM threshold.
--- Firmware/src/modules/sensors/sensors.cpp	2015-04-07 12:50:25.906575935 +0530
+++ pandapilot_v4/src/modules/sensors/sensors.cpp	2015-04-07 16:24:43.358002695 +0530
@@ -122,6 +122,12 @@
 #define ADC_AIRSPEED_VOLTAGE_CHANNEL	15
 #endif
 
+#ifdef CONFIG_ARCH_BOARD_NAVSTIK_V1
+ #define ADC_BATTERY_VOLTAGE_CHANNEL	11
+ #define ADC_BATTERY_CURRENT_CHANNEL  	1
+ #define ADC_AIRSPEED_VOLTAGE_CHANNEL	10
+#endif
+
 #ifdef CONFIG_ARCH_BOARD_AEROCORE
 #define ADC_BATTERY_VOLTAGE_CHANNEL	10
 #define ADC_BATTERY_CURRENT_CHANNEL	-1
@@ -905,12 +911,20 @@
 
 	} else {
 
+#ifdef CONFIG_ARCH_BOARD_NAVSTIK_V1
+
+		/* set the gyro internal sampling rate up to at least 800Hz */
+		ioctl(fd, GYROIOCSSAMPLERATE, 800);
+
+		/* set the driver to poll at 800Hz */
+		ioctl(fd, SENSORIOCSPOLLRATE, 800);
+#else
 		/* set the gyro internal sampling rate to default rate */
 		ioctl(fd, GYROIOCSSAMPLERATE, GYRO_SAMPLERATE_DEFAULT);
 
 		/* set the driver to poll at default rate */
 		ioctl(fd, SENSORIOCSPOLLRATE, SENSOR_POLLRATE_DEFAULT);
-
+#endif
 	}
 
 	return OK;
